src/compiler/scala/tools/nsc/backend/jvm/analysis/BackendUtils.scala

```diff
@@ -263,7 +263,10 @@ class BackendUtils[BT <: BTypes](val btypes: BT) {
 
     def visitInternalName(internalName: InternalName): Unit = if (internalName != null) {
       val t = classBTypeFromParsedClassfile(internalName)
-      if (t.isNestedClass.get) innerClasses += t
+      t.isNestedClass match {
+        case Right(true) => innerClasses += t
+        case _ => // ignores absent classes
+      }
```

Comment:
I would argue that compilation should require this class to be on the classpath and fail otherwise. If we just ignore it here, the resulting classfile is different whether or not something is on the classpath or not, which is problematic IMO.